server/controllers/profile.js

```diff
@@ -0,0 +1,48 @@
+const mongoose = require('mongoose');
+const Profile = require('../models/Profile');
+const asyncHandler = require('express-async-handler');
+
+// POST /profile - Create new profile
+exports.createNewProfile = asyncHandler(async (req, res) => {
+  const profile = req.body;
+  const newProfile = new Profile({ ...profile });
+
+  await newProfile.save();
```

Comment:
You can store the response in a variable and below this you can perform the error handling if there was an error saving the document to the DB.